lib/prometheus_parser.rb

```diff
@@ -33,7 +33,7 @@ def self.parse(raw)
       # https://github.com/rabbitmq/rabbitmq-server/discussions/5143
       value = "NaN" if value == "unknown"
```

Comment:
General remark, might be easier to stick to `float` everywhere, rather than sometimes returning a string and sometimes a float, something like:

```suggestion
      value = Float::NAN if ["unknown", "NaN"].include?(value)
```

Hopefully that serializes the way we need it to serialize later on :)